### ISSUE TITLE
[FEAT]: Exclude list for Team Code Review settings 

### ISSUE THREAD

**Issue by felixlut** [integrations/terraform-provider-github#1972]
### Describe the need

This is a continuation of #1249 which I think should be closed already by #1356

The current version of [`github_team_settings`](https://registry.terraform.io/providers/integrations/github/latest/docs/resources/team_settings) is missing one field that should be available which makes it possible to exclude certain members of the team from being assigned to review PR:s (see image below). 

![image](https://github.com/integrations/terraform-provider-github/assets/34520175/fae3401d-f264-4ffb-bc34-ccb046436239)

The GraphQL API seems to be the only way to programmatically alter these fields, and they are [currently in preview](https://docs.github.com/en/graphql/overview/schema-previews#team-review-assignments-preview). The [`excludedTeamMemberIds`](https://docs.github.com/en/graphql/reference/input-objects#updateteamreviewassignmentinput) list is available in the [`updateTeamReviewAssignment`](https://docs.github.com/en/graphql/reference/mutations#updateteamreviewassignment) mutation call, so the functionality should be possible. 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by felixlut** [comment]
Working on this in #1965, but it's currently semi blocked by the [preview status of the API](https://docs.github.com/en/graphql/overview/schema-previews#team-review-assignments-preview)

**Comment by nunocenteio** [comment]
Can this be reopened, please?

**Comment by nunocenteio** [comment]
Also this setting is very important to control.
 
<img width="549" alt="image" src="https://github.com/user-attachments/assets/95474f47-d2fd-41e9-bdb2-8b4ada4c4561">

**Comment by DaniloGT** [comment]
I need this feature too!

**Comment by janeklb** [comment]
Another attempt at this (courtesy of copilot) https://github.com/integrations/terraform-provider-github/pull/3095

**Comment by xdm67x** [comment]
When I looked at the GraphQL API I did find [this](https://docs.github.com/en/graphql/reference/teams#object-team), but I don't see anything related to the **excluded team members** in the API. You mentioned that it is in preview, do we know when GitHub will implement this in their GraphQL API? It's year 2026 it has been 3 years from the initial request.

**Comment by johanneswuerbach** [comment]
We would also be interested in this. Any chances to raise the missing API internally @stevehipwell ? Happy to create a PR here once there is an API to read the list of excluded members.

**Comment by deiga** [comment]
Looks like the API might support this now: https://docs.github.com/en/enterprise-cloud@latest/graphql/reference/pulls#input-object-updateteamreviewassignmentinput

**Comment by felixlut** [comment]
> Looks like the API might support this now: https://docs.github.com/en/enterprise-cloud@latest/graphql/reference/pulls#input-object-updateteamreviewassignmentinput

No, `UpdateTeamReviewAssignmentInput` has always been there, it's what [my original PR](https://github.com/integrations/terraform-provider-github/pull/1965/files#diff-a9e5227733ba1c9d0e37c953528d541fef9d8c562a0436f1338be15365cd7a9cR199-R205) used. Problem is that it's not possible to read from the API what the current status of this field is. Terraform doesn't like that

https://github.com/integrations/terraform-provider-github/pull/3095#issuecomment-3775238658 basically this

**Comment by stevehipwell** [comment]
FYI I've requested that GitHub look at adding support for this functionality to the REST API.

CC @robert-crandall
